spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/QueryMapper.java

```diff
@@ -302,6 +314,84 @@ private Condition mapCondition(CriteriaDefinition criteria, MapSqlParameterSourc
 				criteria.isIgnoreCase());
 	}
 
+	/**
+	 * Converts values while taking special value types like arrays, {@link Iterable}, or {@link Pair}.
+	 * 
+	 * @param property the property to which the value relates. It determines the type to convert to. Must not be
+	 *          {@literal null}.
+	 * @param value the value to be converted.
+	 * @return a non null {@link JdbcValue} holding the converted value and the appropriate JDBC type information.
+	 */
+	private JdbcValue convertSpecial(RelationalPersistentProperty property, @Nullable Object value) {
```

Comment:
Having `special` in a method name is typically a red flag. How about something like `getWriteValue`?

Reply:
I don't think `getWriteValue` really conveys what the method is doing. And it is handling the special values. 

I agree that the _special_ is suspicious and there might be a bette abstraction hiding or at least a better name, I don't think `getWriteValue` is it.